csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -122,11 +134,6 @@ public BuildAnalysis(Options options, IProgressMonitor progress)
 
             ResolveConflicts();
 
-            if (options.UseMscorlib)
```

Comment:
Why is it not sufficient to just move this codeblock above the call to `ResolveConflicts()`?

Reply:
It might have been enough if we wanted to solve the issue with the single duplicate `System.Private.CoreLib.dll`. I wanted to include the entire folder of that DLL, because it feels somewhat odd to single out the `System.Private.CoreLib.dll`. Maybe most of the DLLs next to it only contain type forwarding attributes, but I think there might be others with valid declarations.

I dug a bit further into some of the available flags, and it seems somewhat hectic what gets added to the references:
- When `ScanNetFrameworkDlls` is set, we try to include all DLLs from certain well-known dotnet locations, such as `/usr/share/dotnet/shared/Microsoft.NETCore.App`.
- If none of them are found, then we add the location of the shipped .net runtime in the codeql CLI.
- If `UseSelfContainedDotnet` is set, then we don't try the well-known dotnet locations, we simply include the shipped .net runtime in the codeql CLI.
- `UseMscorlib` (before this PR) was adding a single DLL, regardless if it was already added or not. 

After the changes in this PR, `UseMscorlib` is pretty much doing what `UseSelfContainedDotnet` is doing. So I think I'll partly revert my change and only cover the deduplication of `System.Private.CoreLib.dll`.